src/ape/api/address.py

```diff
@@ -90,6 +90,22 @@ def balance(self) -> int:
 
         return self.provider.get_balance(self.address)
 
+    # @balance.setter
+    # NOTE: commented out because of failure noted within `__setattr__`
+    def _set_balance_(self, value: Union[str, int]):
+        self.provider.set_balance(
+            self.address,
+            self.conversion_manager.convert(value, int) if isinstance(value, str) else value,
+        )
+
+    def __setattr__(self, attr: str, value: Any) -> None:
```

Comment:
Think it's best not even to bother with typing on this temporary code
```suggestion
    def __setattr__(self, attr, value):
```

Reply:
possible now with pydantic 2
https://github.com/pydantic/pydantic/pull/5502